src/run_model.jl

```diff
@@ -26,6 +26,44 @@ function run_model(::Type{T},P::Parameter) where {T<:AbstractFloat}
 
     @unpack Tprog = P
 
+    G = Grid{T,Tprog}(P)
+    C = Constants{T,Tprog}(P,G)
+    F = Forcing{T}(P,G)
+    # S = ModelSetup{T,Tprog}(P,G,C,F)
+
+    Prog = initial_conditions(Tprog,G,P,C)
+    Diag = preallocate(T,Tprog,G)
+
+    # one structure with everything already inside 
+    S = ModelSetup{T,Tprog}(P,G,C,F,Prog,Diag,0)
+    P = time_integration(S)
+
+    return S, P
+
+end
+
+
+#### Original ##################################################################################
+
+# In order to run the following you need to change lines in model_setup.jl
+
+function mk_run_model(::Type{T}=Float32;     # number format
```

Comment:
Sorry I don't understand how `mk_run_model` is different from `run_model`. I'm happy for you to change the `run_model` function directly, you can provide keyword arguments to return the same values as before (or not)

Reply:
`mk_run_model` left all the structures as is (didn't add `Prog` and `Diag` to `S`). I only added this to check that the integration in both cases returns the same thing (both with single struct `S` and separate structs), I can re-remove it!